lib/cocoapods/installer/xcode/pods_project_generator.rb

```diff
@@ -62,15 +59,15 @@ def initialize(sandbox, aggregate_targets, pod_targets, analysis_result, install
         end
 
         def generate!
-          prepare
-          install_file_references
-          @target_installation_results = install_targets
-          integrate_targets(@target_installation_results.pod_target_installation_results)
-          wire_target_dependencies(@target_installation_results)
-          @target_installation_results
+          project = prepare
+          install_file_references(project)
+          target_installation_results = install_targets(project)
+          integrate_targets(target_installation_results.pod_target_installation_results)
+          wire_target_dependencies(project, target_installation_results)
+          PodsProjectGeneratorResult.new(project, target_installation_results)
         end
 
-        def write
+        def write(project, target_installation_results)
```

Comment:
should these now become class methods?

Reply:
Looks like generator is still using the `installation_options` ivar in this function. We could pass it in from the installer and I think that would allow us to make it a class method. What do you think?

Reply:
I'm actually working on a refactor that will remove the need for `installation_options` in this location - if this PR lands then I can rework this a bit in my refactor

Reply:
@amorde done!